### PR TITLE
Disable and warn user on box creation about no location or product for current base

### PR DESCRIPTION
**Pull request by fhenrich33**

### CODE REVIEW

**Comment by fhenrich33** [comment]
[2.0 UX Bug: When trying to create a box in a base with no instock locations and products, there is no warning to the user.](https://trello.com/c/oB9ZizjC/1446-20-ux-bug-when-trying-to-create-a-box-in-a-base-with-no-instock-locations-and-products-there-is-no-warning-to-the-user)

**Comment by fhenrich33** [comment]
@HaGuesto @pylipp Need help with a sample base with no instock locations or products created to check this out.

**Comment by pylipp** [comment]
> @HaGuesto @pylipp Need help with a sample base with no instock locations or products created to check this out.

@fhenrich33 I think that when this is on staging, you can log in as `some.admin@boxtribute.org` and then create a fresh base in dropapp.

File: front/src/views/BoxCreate/BoxCreateView.tsx [hunk 1]
```diff
@@ -86,6 +88,12 @@ function BoxCreateView() {
   const navigate = useNavigate();
   const { triggerError } = useErrorHandling();
   const { createToast } = useNotification();
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+  const baseName = globalPreferences.selectedBase?.name;
+
+  // no instock warehouse location or products associated with base
+  const [noInstockLocation, setNoInstockLocation] = useState(false);
```
Comment (HaGuesto) [inline front/src/views/BoxCreate/BoxCreateView.tsx:95]: rename to noLocation

File: front/src/views/BoxCreate/BoxCreateView.tsx [hunk 2]
```diff
@@ -142,6 +150,16 @@ function BoxCreateView() {
     }))
     .sort((a, b) => Number(a?.seq) - Number(b?.seq));
 
+  useEffect(() => {
+    // Reset on query
+    setNoInstockLocation(false);
```
Comment (HaGuesto) [inline front/src/views/BoxCreate/BoxCreateView.tsx:155]: pls remove

File: front/src/views/BoxCreate/BoxCreateView.tsx [hunk 3]
```diff
@@ -222,27 +240,36 @@ function BoxCreateView() {
...
-  if (qrCodeExists.loading || allFormOptions.loading || createBoxMutationState.loading) {
+  if (qrCodeExists.loading || allFormOptions.loading || createBoxMutationState.loading)
     return <APILoadingIndicator />;
-  }
 
   // TODO: handle errors not with empty div, but forward or roll data back in the view
   if (
     !qrCodeExists.data?.qrExists ||
     qrCodeExists.error ||
     allLocations === undefined ||
```
Comment (HaGuesto) [inline front/src/views/BoxCreate/BoxCreateView.tsx:250]: replace the case with a error state of allFormOptions or !allFormOption.data.base is undefined.

File: front/src/views/BoxCreate/BoxCreateView.tsx [hunk 4]
```diff
@@ -222,27 +240,36 @@ function BoxCreateView() {
...
     allProducts === undefined
-  ) {
+  )
     return <div />;
```
Comment (HaGuesto) [inline front/src/views/BoxCreate/BoxCreateView.tsx:253]: add an alert here with the message: "Could not fetch QR, Location and Product data! Please try reloading the page."

**Review by HaGuesto — COMMENTED** [review]
change:
- remove the filter for BoxState.Scrap in line 145 since we have users creating boxes actually in a scrap location.

add:
- tests for alerts

File: front/src/views/BoxCreate/BoxCreateView.tsx (hunk 3 above)
Comment (fhenrich33) [reply front/src/views/BoxCreate/BoxCreateView.tsx:250]: Should this be for the whole if statement?

Comment (HaGuesto) [reply front/src/views/BoxCreate/BoxCreateView.tsx]: remove the cases for `allLocations === undefined` and `allProducts === undefined`

**Review by HaGuesto — APPROVED** [review]
# code review and functional review passed

I fixed the small issues I saw myself @fhenrich33

**Comment by HaGuesto** [comment]
@fhenrich33 there is one test failing. Could you check it?

**Comment by fhenrich33** [comment]
> @fhenrich33 there is one test failing. Could you check it?

@HaGuesto this is one of the flaky ones.


    Flaky test files:
    front/src/views/Transfers/ShipmentView/ShipmentView.test.tsx
    front/src/views/Transfers/CreateTransferAgreement/CreateTransferAgreementView.test.tsx
    front/src/views/Boxes/BoxesView.test.tsx
    front/src/views/QrReader/components/ResolveHash.test.tsx

**Comment by fhenrich33** [comment]
> @fhenrich33 However, since the resolveHash test has something to do with the createBox flow and the test is continuously failing, I'd like you to take a look at it.

If you check out other CI runs, you will see that this is flaky in CI only, and there's nothing I can do about that, at least until I handle this with the test suite revamp with Playwright, which I'm doing right now.

**Comment by fhenrich33** [comment]
@HaGuesto From #1722 

https://app.circleci.com/pipelines/github/boxwise/boxtribute/6057/workflows/6d69ee2e-fcb9-42ed-ae0d-4b0a706259b2/jobs/34305

**Pull request merged**